### ISSUE TITLE
Broken links in generated "API reference"

### ISSUE THREAD

**Issue by Alexqwesa** [dart-lang/pub-dev#7058]
If you open link below, left-side menu contains section `CONSTRUCTORS` - all links in this section are broken:
https://pub.dev/documentation/locale_switcher/latest/locale_switcher/LocaleSwitcher/LocaleSwitcher.iconButton.html

Actual links look like this:
https://pub.dev/documentation/locale_switcher/latest/locale_switcher/locale_switcher/LocaleSwitcher/LocaleSwitcher.menu.html

But they should look like this: 
https://pub.dev/documentation/locale_switcher/latest/locale_switcher/LocaleSwitcher/LocaleSwitcher.menu.html

**Comment by parlough** [comment]
This looks related to https://github.com/dart-lang/dartdoc/commit/2ae78bcc2f05d85cad0e861ef560f820a9c6fa87 which split out sidebars into separate HTML files to reduce the size of built documentation.

Without looking closely, I think it's likely an issue with pub's custom setup not being updated to handle the change rather than the dartdoc change itself though.

\cc @srawlins Just so you're aware.

**Comment by isoos** [comment]
I think the root cause can be seen on these two pages:
https://pub.dev/documentation/locale_switcher/latest/locale_switcher/LocaleSwitcher-class.html
https://pub.dev/documentation/locale_switcher/latest/locale_switcher/LocaleSwitcher/LocaleSwitcher.custom.html

Both of them load the `locale_switcher/LocaleSwitcher-class-sidebar.html`, the first one on the right/below side, the second one on the left/above side. Because it loads the same file from a different directory, the `../*` relative links won't work in one of them, while it will work in the other.

I think this may be a bug in dartdoc (e.g. we may need two version of the sidebars, one for the left and one for the right side), but I'm unsure, we may be doing the inclusions wrong. We may also override the sidebar html in a way that creates absolute URLs, so it will work on both sides.

**Comment by srawlins** [comment]
This appears to be for leaf nodes of every single package on pub?

I will do some more testing locally but I thought I was pretty careful with my change, and I think the pub processing does do some URL rewriting...

**Comment by isoos** [comment]
> This appears to be for leaf nodes of every single package on pub?

Every second-level (`https://pub.dev/documentation/<pkg>/<version>/<library>/<class>/<field>.html`) is affected. The same file is included in the first-level, and at level it is working ok.

> I will do some more testing locally but I thought I was pretty careful with my change, and I think the pub processing does do some URL rewriting...

By any means, please let us know what's different, we may be doing (or we may be not doing) something that breaks it.

**Comment by isoos** [comment]
Difference between pub.dev and local output:
https://pub.dev/documentation/pana/latest/models/AnalysisResult/contributingUrl.html
<img width="772" alt="image" src="https://github.com/dart-lang/pub-dev/assets/4778111/8aebd1bd-c083-4b71-80bb-012742c093e0">

That marked content on the left-side is not present when the generated content is served locally. However, the HTML content has the same `data-above-sidebar="models/AnalysisResult-class-sidebar.html"` which in other cases triggers the inclusion of the sidebar, so not sure if the sidebar didn't get loaded properly or by intent. Any thought on that @srawlins

**Comment by parlough** [comment]
As an update for those following, the fixes for both of the above issues have landed and they worked well when testing locally (based on `package:dartdoc` v7.0.0). Thanks @srawlins :D 

I was going to open a PR to update `pub-dev`, but I discovered the fix introduced a small regression. It accidentally overwrites some of the content on the narrow screen sidenav, including the search bar. I created a fix at https://github.com/dart-lang/dartdoc/pull/3525 if it makes sense to wait for that and a 7.0.1 release.
